hr_period/models/hr_payslip.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding:utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2015 Savoir-faire Linux. All Rights Reserved.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning as UserError
+
+
+class HrPayslip(models.Model):
+    _inherit = 'hr.payslip'
+
+    hr_period_id = fields.Many2one(
+        'hr.period',
+        string='Period',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    date_payment = fields.Date(
+        'Date of Payment',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+
+    @api.multi
+    @api.constrains('hr_period_id', 'company_id')
+    def _check_period_company(self):
+        for slip in self:
+            if slip.hr_period_id:
+                if slip.hr_period_id.company_id != slip.company_id:
+                    raise UserError("The company on the selected period must "
+                                    "be the same as the company on the "
+                                    "payslip.")
+        return True
+
+    @api.onchange('company_id', 'contract_id')
+    def onchange_company_id(self):
+        if len(self.company_id) and len(self.contract_id):
+            contract = self.contract_id
+            period_obj = self.env['hr.period']
+            period = period_obj.get_next_period(self.company_id.id,
+                                                contract.schedule_pay)
+            self.hr_period_id = period.id if period else False
+
+    @api.onchange('contract_id')
+    def onchange_contract_id(
+        self, cr, uid, ids, date_from, date_to,
```

Comment:
Use the new api here. I know that the super method uses the old api, but you still can remove cr, uid, ids, context and use api.multi instead. You must either use api.onchange or the deprecated on_change attribute inside the form view.